src/net.h

```diff
@@ -236,6 +236,10 @@ class CNetMessage {
     std::string m_type;
 
     CNetMessage(CDataStream&& recv_in) : m_recv(std::move(recv_in)) {}
+    CNetMessage(CNetMessage&&) = default;
```

Comment:
Done! happy to amend the comment if you had something different in mind.

Reply:
This is absolutely fine. I would have gone for something like "For performance reasons, once we've deserialized the bytes into a CNetMessage, we avoid re-allocating and copying the message as we pass it up the stack for processing. Delete the copy ctor/assignment to avoid accidentally introducing a copy operation.", which is more-or-less the same as what you came up with.